pkg/platform/api/buildplanner/request/attachStagedCommit.go

```diff
@@ -0,0 +1,197 @@
+package request
+
+func AttachStagedCommit(owner, project, parentCommit, stagedCommit, branch string) *buildPlanByAttachStagedCommit {
+	return &buildPlanByAttachStagedCommit{map[string]interface{}{
+		"organization": owner,
+		"project":      project,
+		"parentCommit": parentCommit,
+		"stagedCommit": stagedCommit,
+		"branchRef":    branch,
+	}}
+}
+
+type buildPlanByAttachStagedCommit struct {
+	vars map[string]interface{}
+}
+
+func (b *buildPlanByAttachStagedCommit) Query() string {
+	return `
+mutation ($organization: String!, $project: String!, $parentCommit: ID!, $stagedCommit: ID!, $branchRef: String!) {
+  attachStagedCommit(input: {organization: $organization, project: $project, parentCommitId: $parentCommit, stagedCommitId: $stagedCommit, branchRef: $branchRef}) {
```

Comment:
Scott suggested to rename this to `mergeCommit` actually (https://activestatef.atlassian.net/browse/PB-4375?focusedCommentId=43427), but it sounds like you also need an API to attach a staged commit to a branch (and possibly even create that branch/project?)

I wonder if that is actually more of a `createProject` use-case... Where you create a project with an existing commit.

Reply:
Also:  You'll need this for the next state tool release, correct?

Reply:
`createProject` might be a useful mutation to add... we have this story: https://activestatef.atlassian.net/browse/DX-2173 which could spawn more use-cases for alternate mutations beyond the ones we have now.